app/routes/category.$categorySlug/route.tsx

```diff
@@ -61,10 +71,38 @@ export const loader = async ({ params, request }: LoaderFunctionArgs) => {
 export default function ProductsCategoryPage() {
     const { categoryProducts, category, allCategories, productPriceBounds } = useLoaderData<typeof loader>();
 
+    const [searchParams] = useSearchParams();
     const { appliedFilters, someFiltersApplied, clearFilters, clearAllFilters } = useAppliedProductFilters();
 
     const currency = categoryProducts.items[0]?.priceData?.currency ?? 'USD';
 
+    const [products, setProducts] = useState(categoryProducts.items);
```

Comment:
we need to move all the pagination logic into separate hook so it can be moved into `lib` folder (when we add it here)